### PR TITLE
[FIVE-163 Controller + ProjectUpsertDTO] Refactorizar el código de los controladores y servicios de proyecto

### PR DESCRIPTION
**Pull request by gonzalorom**
Se debe refactorizar el código de los controladores y los servicios para:
Asegurarse que todos utilicen métodos Asíncronos
Agregar el sufijo Async en el nombre de todos los métodos asíncronos
Agregar DTO’s de entrada de datos

Cambios Realizados:
- Se creó ProjectUpsertDTO.
- Se cambio naming de ProjectDto a ProjectDTO para mantener estilo.
- Se actualizó ProjectController:
	- Se agrega sufijo Async a los endpoints.
	- Se cambian respuestas en varios endpoints.

### CODE REVIEW

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs [hunk 1]
```diff
@@ -40,97 +41,69 @@ public async Task<IActionResult> GetAll(string userId)
...
-            
-                        var projectsDto = _mapper.Map<IEnumerable<ProjectDto>>(projects);
-                        return Ok(projectsDto);*/
-            return NotFound();
+            // TODO: AWS Credentials, Loggin bypassed.Uncomment after do:
+            //var currentUserId = await _userService.GetCurrentUserId();
+            var currentUserId = "";
+            var projects = await _projectService.GetAllAsync(currentUserId);
+
+            if (projects == null) return StatusCode(204);
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:56]: Same as in the other PR. Let's talk about this. What is the service returning? Null or empty list?

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs [hunk 2]
```diff
@@ -40,97 +41,69 @@ public async Task<IActionResult> GetAll(string userId)
...
         }
 
         [HttpGet("{id}")]
-        public async Task<IActionResult> Get(int id)
+        public async Task<IActionResult> GetAsync(int id)
         {
-            //TODO: AWS Credentials, Loggin bypassed. Uncomment after do:
-            //var userId = await _userService.GetCurrentUserId();
             var project = await _projectService.GetAsync(id);
+            if (project == null) return NoContent();
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:66]: NotFound

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs [hunk 3]
```diff
@@ -40,97 +41,69 @@ public async Task<IActionResult> GetAll(string userId)
...
-            }
-
-            var project = await _projectService.GetAsync(projectDto.Id);
-
-            if (project == null)
-            {
-                return NotFound();
-            }
+            var project = await _projectService.GetAsync(id);
+            if (project == null) return NoContent();
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:89]: NotFound

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs [hunk 4]
```diff
@@ -40,97 +41,69 @@ public async Task<IActionResult> GetAll(string userId)
...
-                return NotFound();
-            }
+            if (project == null) return NotFound();
 
             var isDeleted = await _projectService.DeleteAsync(id);
-            if (!isDeleted)
-            {
-                return NotFound();
-            }
+            if (!isDeleted) return NotFound();
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:106]: Bad request maybe? It's debatable.

File: FiveRockingFingers/FRF.Web/Controllers/UserController.cs [hunk 5]
```diff
@@ -79,16 +80,39 @@ public async Task<IActionResult> SearchUserAsync(string email)
...
+            /*var userId = await _userService.GetUserIdByEmail(email);
+            if (userId == null) return NotFound();
+
+            var userPublicProfile = new UserPublicProfileDTO
+            {
+                FullName = await _userService.GetFullname(email),
+                UserId = userId,
+                Email = email,
+                Avatar = null
+            };*/
```
Comment (matiasbeckerle) [inline FiveRockingFingers/FRF.Web/Controllers/UserController.cs:115]: These changes are needed in this PR? Seems unrelated.

**Review by matiasbeckerle — COMMENTED** [review]
Waaay better PR. Nice work. Let's tune/talk the response types before merging.

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs (hunk 1 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:56]: You are right. Here I'll change it for `if(!projects.Any()) return StatusCode(204);` because the service is returning a empty list.

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs (hunk 2 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:66]: Same but now `if(!projects.Any()) return NotFound;`

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs (hunk 1 above)
Comment (matiasbeckerle) [reply FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs:56]: If that's the case then I will advice returnin 200 OK with an empty list. So you don't need to add any special lines...

```
var projectsDto = _mapper.Map<List<ProjectDTO>>(projects);
return Ok(projectsDto);
```

should be enough for both scenarios (empty or not empty).

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs (hunk 3 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs]: Done.

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs (hunk 2 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs]: Done.

File: FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs (hunk 1 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/Controllers/ProjectsController.cs]: Yes, that way looks neat.

File: FiveRockingFingers/FRF.Web/Controllers/UserController.cs (hunk 5 above)
Comment (gonzalorom) [reply FiveRockingFingers/FRF.Web/Controllers/UserController.cs]: I took it out.

**Review by matiasbeckerle — APPROVED** [review]
